doc/adr/0006-explicit-resources-for-declaring-service-bindings.md

```diff
@@ -0,0 +1,61 @@
+# 6. Explicit resources for declaring service bindings
+
+Date: 2024-01-16
+
+## Status
+
+Proposed
+
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
+
+## Context
+
+Service bindings are the configurations that determine how a Skupper
+service is invoked (ingress binding) and what processes that
+invocation may be routed to (egress binding).
+
+The Deployments or StatefulSets that are to be annotated may
+themselves by the generated output of other controllers. In this case
+it is not always possible to ensure that they are annotated correctly
+for Skupper.
+
+The way annotations work when applied to Service resources is
+confusing and unintuitive.
+
+If the skupper.io/proxy annotation is found on a Service, the egress
+binding is configured based on the original selector in the Service
+and this selector is then changed to point the service at the router
+pods. This changing of the original resource can cause problems when a
+GitOps approach is used for deployment.
+
+However if the skupper.io/target annotation is also applied to the
+Service, the behaviour is different. Here the annotated service is
+updated to point at the skupper router pods, and the value of target
+is used to configure the egress binding.
+
+If instead the skupper.io/ingress-only annotation is also present on
+the Service, no target is inferred. The service is configured to point
+at the router pods on a port that will route to the address specified
+by the skupper.io/address annotation.
+
+The code accummulated to handle all these configurations as well as
+the CLI is not the easiest to maintain.
+
+## Decision
+
+Explicit resources will be introduced that describe the desired
+configuration for both ingress- and egress- bindings.
+
```

Comment:
Suggest detailing the resource definition configuration (even if example only to be refined later)

Reply:
To my way of thinking, the details of the resource definitions to be used is a distinct decision from that of requiring explicit resources of some kind. I wrote this ADR with the view that a decision on the principle was important before trying to get a decision on the details and felt that including details would confuse that.

Happy to revisit that view if it is not shared, but just clarifying my thinking.